java/client/src/main/java/glide/managers/CommandManager.java

```diff
@@ -29,83 +29,203 @@ public class CommandManager {
     /**
      * Build a command and send.
      *
-     * @param command The command to execute
+     * @param requestType Redis command type
+     * @param arguments Redis command arguments
+     * @param route Command routing parameters
      * @param responseHandler The handler for the response object
      * @return A result promise of type T
      */
     public <T> CompletableFuture<T> submitNewCommand(
-            Command command, RedisExceptionCheckedFunction<Response, T> responseHandler) {
-        // write command request to channel
-        // when complete, convert the response to our expected type T using the given responseHandler
-        return channel
-                .write(
-                        prepareRedisRequest(
-                                command.getRequestType(),
-                                command.getArguments(),
-                                Optional.ofNullable(command.getRoute())),
-                        true)
-                .thenApplyAsync(responseHandler::apply);
+            RequestType requestType,
+            String[] arguments,
```

Comment:
This is passed in from Options.  Which builds a list and sends that back.  
I think this is fine as is, because we don't need the lists to be mutable.  The only time this is kind of "problem" is in tests when we want to check all the args in the protobuf message and it's sometimes annoying to build arrays.  But this is really nit/minor.